File: Lab5/C++/lab5.2/Number.cpp
#include "Number.h"

Number::Number() : Str() { };
Number::Number(string s) { str = s; LenCount(s); };

void Number::LenCount(string s) 
{
	int tempLen = 0;
	error = false;
	for (int i = 0; i < s.length(); i++) 
	{		
		if (isdigit(s[i])) {
			tempLen++;
		}
		else error = true;
	}
	len = tempLen;
}
void Number::Processing()
{
	for (int i = 0; i < str.length(); i++) {
		if (isdigit(str[i])) 
		{			
			str.insert(i, 1, str[i]);
			i++;
		}
	}
	LenCount(str);
}

File: Lab5/C#/lab5.2/Number.cs
using System;


namespace lab5._2
{
    class Number: Str
    {
        public Number() : base() { }
        public Number(string s) : base(s) { }

        public override void Lenght()
        {
            int numbersInStr = 0;
            for (int i = 0; i < str.Length; i++)
            {
                if (Char.IsNumber(str[i]))
                {
                    numbersInStr++;
                }
                else { error = true; }
            }
            Len = numbersInStr;
        }

        public override void Processing()
        {
            for (int i = 0; i < str.Length; i++)
            {
                if (Char.IsNumber(str[i]))
                {
                    str = str.Insert(++i, str[i-1].ToString());
                }
            }
            this.Lenght();
        }
    }
}
